Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs

```diff
@@ -643,6 +643,44 @@ public void EnumerateDirectoryViaDirectorySymlinkShouldBeObservedAsDirectoryEnum
             RunScheduler().AssertCacheMiss(processWithOutputs.Process.PipId);
         }
 
+        /// <summary>
+        /// Creates a shared opaque directory producer & consumer and verifies their usage and caching behavior
+        /// </summary>
+        [FactIfSupported(requiresSymlinkPermission: true)]
+        public void PipDependsOnFileAndSourceFileAlternatePathUnderDirectorySymlink()

```

Comment:
>PipDependsOnFileAndSourceFileAlternatePathUnderDirectorySymlink [](start = 20, length = 63)

I'm not so sure how this test relates to your change or to the scenario we're dealing with. It doesn't seem to introduce any directory symlink. Also, the change only takes effect if the process has `IsDistributedWorker`, no?